crate_universe/src/metadata/cargo_tree_resolver.rs

```diff
@@ -503,18 +697,74 @@ impl Ord for DependencyDetailWithOrd {
 
 impl Eq for DependencyDetailWithOrd {}
 
+/// A wrapper for [CrateId] used by [parse_cargo_tree_output] to successfully
+/// parse target and host dependencies.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+enum TreeDepCompileKind {
+    /// Collecting dependencies for the target platform.
+    Target(CrateId),
+
+    /// Collecting dependencies for the host platform (e.g. `[build-dependency]` and `proc-macro`).
+    Host(CrateId),
+
+    /// Similar to [TreeDepCompileKind::Target] represents an edge where
+    /// dependencies are to be collected for host. E.g. a crate which has
+    /// a `[build-dependency] or `(proc-macro)` dependency.
+    TargetWithHostDep(CrateId),
```

Comment:
I'm not quite clear on what these deps are, the docs read that they're the same as `Host`? Can you try clarifying the docs?

Reply:
Added some more words 😄